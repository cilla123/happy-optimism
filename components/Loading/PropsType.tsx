import PropsType from '../Toast/PropsType';

declare global {
  interface Window {
    zarmLoading?: any;
  }
}

export default PropsType;
