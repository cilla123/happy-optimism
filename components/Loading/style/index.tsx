import '../../style/';
import '../../Toast/style/';
import '../../Spinner/style/';
import './index.scss';
