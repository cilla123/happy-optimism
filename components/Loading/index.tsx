import React, { PureComponent } from 'react';
import ReactDOM from 'react-dom';
import PropsType from './PropsType';
import Toast from '../Toast';
import Spinner from '../Spinner';

export interface LoadingProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Loading extends PureComponent<LoadingProps, {}> {

  static defaultProps = {
    prefixCls: 'za-loading',
  };

  static show = (props) => {
    ReactDOM.render(<Loading {...props} visible />, window.zarmLoading);
  }

  static hide = () => {
    ReactDOM.render(<Loading visible={false} />, window.zarmLoading);
  }

  render() {
    const { prefixCls, ...others } = this.props;

    return (
      <Toast prefixCls={prefixCls} {...others} stayTime={0}>
        <Spinner size="lg" className="rotate360" />
      </Toast>
    );
  }
}

if (!window.zarmLoading) {
  window.zarmLoading = document.createElement('div');
  document.body.appendChild(window.zarmLoading);
}

ReactDOM.render(<Loading visible={false} />, window.zarmLoading);
