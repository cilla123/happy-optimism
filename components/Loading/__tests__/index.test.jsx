import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from '../index';

describe('Confirm', () => {
  it('renders correctly', () => {
    const wrapper = render(<Loading visible />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
