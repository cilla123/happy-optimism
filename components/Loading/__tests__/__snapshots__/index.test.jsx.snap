// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm renders correctly 1`] = `
<div
  class="za-loading za-loading-open"
>
  <div
    class="za-loading-container"
  >
    <svg
      class="za-spinner rotate360 theme-primary size-lg"
      viewBox="0 0 62 62"
    >
      <circle
        class="za-spinner-path"
        cx="31"
        cy="31"
        fill="none"
        r="28.5"
        style="stroke-width:5"
      />
      <circle
        class="za-spinner-line"
        cx="31"
        cy="31"
        fill="none"
        r="28.5"
        style="stroke-dasharray:26.860617188192727 179.0707812546182;stroke-width:5"
      />
    </svg>
  </div>
</div>
`;
